client/grpc/options.go

```diff
@@ -72,3 +75,27 @@ func MaxSendMsgSize(s int) client.Option {
 		o.Context = context.WithValue(o.Context, maxSendMsgSizeKey{}, s)
 	}
 }
+
+//
+// DialOptions to be used to configure gRPC dial options
+//
+func DialOptions(opts ...grpc.DialOption) client.Option {
+	return func(o *client.Options) {
+		if o.Context == nil {
+			o.Context = context.Background()
+		}
+		o.Context = context.WithValue(o.Context, grpcDialOptions{}, opts)
+	}
+}
+
+//
+// CallOptions to be used to configure gRPC call options
+//
+func CallOptions(opts ...grpc.CallOption) client.Option {
```

Comment:
This should likely return a client.CallOption

Reply:
Solved at https://github.com/micro/go-micro/pull/833/commits/77108771db383631df49ef08732eb5159b85157c#diff-4c65aaabc8f3a5b0c00b82ff6d39a485R94

Reply:
Here don't work, because the `client.Option` apply on the copy of grpcClient.opts.CallOptions.Context, and `g.getGrpcDialOptions()` get `grpc.CallOption` from origin grpcClient.opts.CallOptions.Context